### PR TITLE
use skipif for optional dependencies

### LINKED ISSUE #67: skip tests dependent on optional packages
In #62, we started using `@pytest.mark.skipif` to skip tests if an optional dependency is not installed. #62 only did this for `pygmsh` and `meshio`. As @mancellin [pointed out](https://github.com/mancellin/capytaine/pull/62#discussion_r618469118), we should also do this for:

- `vtk` (`test_ui_vtk.py` and `test_io_meshes.py`)
- `matplotlib` (`test_matrices.py`)

### PR DESCRIPTION
**Pull request by ryancoe**
fixes #67

### CODE REVIEW

**Comment by ryancoe** [comment]
There are many `asserts` called within a single test function in `test_matrices.py`... I think it would probably be best practice to break these out to be separate tests (can use a `@pytest.fixture` to do the set up). This would also allow us to use the `@pytest.skipif` decorator to skip the `matplotlib` dependent tests. Are you ok with this approach @mancellin (i.e., breaking out all these asserts to separate tests)?

https://github.com/mancellin/capytaine/blob/87d09bf56dcd1d4a242a29805e9d49511b3947e1/pytest/test_matrices.py#L18-L62

**Comment by mancellin** [comment]
> I think it would probably be best practice to break these out to be separate tests

To identify more easily the failing assertion? It makes sense, that's ok with me.
Although, you seem to have two tests with the same name in bf535fa, is it ok with pytest?

**Comment by ryancoe** [comment]
**two tests with same name:** No, I don't think that's OK (good catch!). Fixed with a7520e1808fee7b43d02ec8db87b95b1b848f598

**break out tests:** with bf535fa, I did this just to the minimum level needed to use `@pytest.mark.skipif`. It can be done more (i.e., so that every `assert` statement is its own test. I don't really see to much value in this though.

**Comment by mancellin** [comment]
All right.

The main need for `skipif` might actually be the tests that require quadpy: they could be skipped if the correct version (0.14) is not installed. Cf #48.

File: pytest/test_bem_with_quadratures.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 
+import pytest
 import numpy as np
 import xarray as xr
 import capytaine as cpt
+
 import quadpy
+from packaging import version
+quadpy_version = version.parse(quadpy.__version__)
+if not quadpy_version <= version.parse('0.14'):
```
Comment (mancellin) [inline pytest/test_bem_with_quadratures.py:12]: Maybe we should test both if quadpy is installed and if the version is correct?

Comment (ryancoe) [reply pytest/test_bem_with_quadratures.py:12]: Right, so we need an `and` statement

Comment (ryancoe) [reply pytest/test_bem_with_quadratures.py]: 0e9d111

**Comment by mancellin** [comment]
Thank you!

**Pull request merged**